### ISSUE TITLE
Thingspeak Read Last Value Fail

### ISSUE THREAD

**Issue by anthonyjclarke** [ThingPulse/esp8266-weather-station#134]
8<------------------------ BUG REPORT -----------------------------------------
### Expected behavior

Used to read last values from Thingspeak

### Actual behavior

Returning Blank values

```
Now reading Ext Temp from Thingspeak

Requesting URL: /channels/294484/feeds.json?results=1&api_key=DLBZRS1W9WYYBP5P
.
Ext Temp =  - Humid = 
```

### Test code

```c
// add code here


const String THINGSPEAK_CHANNEL_ID = "294484";
const String THINGSPEAK_API_READ_KEY = "DLBZRS1W9WYYBP5P";

// These are Public - I'm ok to share :) just the temperature on my side deck on the house!

thingspeak.getLastChannelItem(THINGSPEAK_CHANNEL_ID, THINGSPEAK_API_READ_KEY);
  ext_temperature = thingspeak.getFieldValue(0);
  ext_humidity = thingspeak.getFieldValue(1);
  
  Serial.println("Ext Temp = "+ String(ext_temperature) +" - Humid = "+String(ext_humidity));
```

### Weather Station version
Which branch are you on? If you know the Git revision then add it here as well.

### Hardware
Describe whether you run the code on the ThingPulse Weather Station Color kit or on some custom hardware.

8<------------------------ END BUG REPORT -------------------------------------

**Comment by Matz88** [comment]
same issue here, TSclient stopped working...

**Comment by marcelstoer** [comment]
Instead we should maybe use the official lib https://github.com/mathworks/thingspeak-arduino

**Comment by Matz88** [comment]
No...the official library is not a good choice because it retrieves only one field per connection then disconnects therefore the download time became worst case 8 times longer...

**Comment by karmacoma92** [comment]
Same issue, ....
Let me know if you need any further testing

**Comment by Matz88** [comment]
I have temporarily solved not keeping the cycle till the client is connected, see the row commented below, from ThingspeakClient.cpp

//while(client.connected()) {
    while((size = client.available()) > 0) {
      c = client.read();
      if (c == '{' || c == '[') {
        isBody = true;
      }
      if (isBody) {
        parser.parse(c);
      }
    }
  //}

It is working but it is not the best because now if the download from TS is faster than the parsing with Jason, it is not completed and part of the value can be not updated... I would like to understand the reason why it is working only commenting out this check. Maybe TS has changed something...

**Comment by marcelstoer** [comment]
Ahh, I see...The TS client is affected by the same as this 038c4dacd838bf7819706b2b0cc0d5229b3ff475, read up on #140 for details.

**Comment by Matz88** [comment]
I will try this fix too good to know, thanks

**Comment by marcelstoer** [comment]
Just pushed a commit that fixes this for all files that were'n included in the previous commit: dddb014f7f9bc14ca982099c1b73d1b4cf8ebc79

Will be in the next release (1.6.4)
